test/foundry/library/LibERC20Transfer.t.sol

```diff
@@ -0,0 +1,95 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+pragma solidity ^0.8.22;
+
+import {IERC20, ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20Errors} from "@openzeppelin/contracts/interfaces/draft-IERC6093.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+import {LibERC20Transfer} from "contracts/library/LibERC20Transfer.sol";
+
+import {TestBase} from "../util/TestBase.sol";
+
+contract NormalToken is ERC20 {
+    constructor(
+        address tokenOwner,
+        uint256 initialSupply
+    ) ERC20("NormalToken", "NORMAL") {
+        _mint(tokenOwner, initialSupply);
+    }
+}
+
+contract LibERC20TransferTest is TestBase {
+    using LibERC20Transfer for IERC20;
+    IERC20 _token;
+
+    function setUp() public {
+        _token = IERC20(_newAddr());
+    }
+
+    function testNoContractCode(address to, uint256 value) public {
+        vm.assume(!_isContract(address(_token)));
+
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                Address.AddressEmptyCode.selector,
+                address(_token)
+            )
+        );
+        _token.safeTransfer(to, value);
+    }
+
+    function testReturnFalse(address to, uint256 value) public {
+        vm.mockCall(
+            address(_token),
+            abi.encodeCall(_token.transfer, (to, value)),
+            abi.encode(false)
+        );
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                SafeERC20.SafeERC20FailedOperation.selector,
+                address(_token)
+            )
+        );
+        _token.safeTransfer(to, value);
+    }
+
+    function testNormalToken(uint256 supply, address to, uint256 value) public {
+        vm.assume(to != address(0));
+        supply = bound(supply, 1, type(uint256).max - 1);
+        value = bound(value, 0, supply);
+        _token = new NormalToken(address(this), supply);
```

Comment:
You could define a `_tokenOwner` state variable and assign it to `_newAddr` in the `setUp` function.

Reply:
Yes. Although since this var is only used in this test. I will place the var in the current test instead of `setUp` function :)